src/YDialogSpy.cc

```diff
@@ -278,70 +474,427 @@ void fillTree( YWidgetTreeItem * 		parent,
     for ( YWidgetListConstIterator it = begin; it != end; ++it )
     {
 	YWidget * widget = *it;
-	YWidgetTreeItem * item = new YWidgetTreeItem( parent, widget, treeLevel < 4 );
+	auto item = new YWidgetTreeItem( parent, widget, treeLevel < 4 );
 
 	if ( widget->hasChildren() )
 	    fillTree( item, widget->childrenBegin(), widget->childrenEnd(), treeLevel+1 );
     }
 }
 
-
+/**
+ * The main loop of the spy dialog
+ */
 void YDialogSpy::exec()
 {
     YUI_CHECK_PTR( priv->spyDialog );
 
     while ( true )
     {
-	bool updateProp = false;
-	YEvent * event = priv->spyDialog->waitForEvent();
-	yuiMilestone() << "dialog: " << priv->spyDialog->preferredHeight();
-	yuiMilestone() << "tree: " << priv->widgetTree->preferredHeight();
+        auto event = priv->spyDialog->waitForEvent();
+        yuiMilestone() << "event: " << event;
+        if (!event) continue;
+
+        // window manager "close window" button
+	    if ( event->eventType() == YEvent::CancelEvent ) break;
```

Comment:
Um, in the editor it looks OK, there is some `<Space>` and `<Tab>` mixture in the indentation which GitHub does not handle well...